x-pack/plugins/ml/public/alerting/ml_anomaly_alert_trigger.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useEffect, useMemo } from 'react';
+import { EuiSpacer, EuiForm } from '@elastic/eui';
+import { JobSelectorControl } from './job_selector';
+import { useMlKibana } from '../application/contexts/kibana';
+import { jobsApiProvider } from '../application/services/ml_api_service/jobs';
+import { HttpService } from '../application/services/http_service';
+import { SeverityControl } from './severity_control';
+import { ResultTypeSelector } from './result_type_selector';
+import { alertingApiProvider } from '../application/services/ml_api_service/alerting';
+import { PreviewAlertCondition } from './preview_alert_condition';
+import { ANOMALY_THRESHOLD } from '../../common';
+import { MlAnomalyDetectionAlertParams } from '../../common/types/alerts';
+
+interface MlAnomalyAlertTriggerProps {
+  alertParams: MlAnomalyDetectionAlertParams;
+  setAlertParams: <T extends keyof MlAnomalyDetectionAlertParams>(
+    key: T,
+    value: MlAnomalyDetectionAlertParams[T]
+  ) => void;
+  setAlertProperty: (key: string, value: any) => void;
+}
+
+const MlAnomalyAlertTrigger: FC<MlAnomalyAlertTriggerProps> = ({
+  alertParams,
+  setAlertParams,
+  setAlertProperty,
+}) => {
+  const {
+    services: { http },
+  } = useMlKibana();
+  const mlHttpService = useMemo(() => new HttpService(http), [http]);
+  const adJobsApiService = useMemo(() => jobsApiProvider(mlHttpService), [mlHttpService]);
+  const alertingApiService = useMemo(() => alertingApiProvider(mlHttpService), [mlHttpService]);
+
+  const onAlertParamChange = useCallback(
+    <T extends keyof MlAnomalyDetectionAlertParams>(param: T) => (
+      update: MlAnomalyDetectionAlertParams[T]
+    ) => {
+      setAlertParams(param, update);
+    },
+    []
+  );
+
+  useEffect(function setDefaults() {
+    if (alertParams.severity === undefined) {
+      onAlertParamChange('severity')(ANOMALY_THRESHOLD.CRITICAL);
+    }
+    if (alertParams.tags === undefined) {
+      setAlertProperty('tags', ['ml']);
+    }
+    return () => {
+      // Reset alert properties on unmount
+    };
+  }, []);
+
+  return (
+    <EuiForm data-test-subj={'mlAnomalyAlertForm'}>
```

Comment:
Is it possible to disable the flyout Save button if the result type or job(s) have not been set?

![image](https://user-images.githubusercontent.com/7405507/107031232-c0665d00-67a9-11eb-879a-2053635d965d.png)

Reply:
This was a decision on the alert form itself to make the Save button active by default: https://github.com/elastic/kibana/issues/84091, which aligns with EUI guidelines. If the result type/jobs are not selected on Save, an error message should be shown.

Reply:
Added client-side validation here https://github.com/elastic/kibana/pull/89286/commits/886c09aee2abe757f03d9d4d39f91cd0982512c1